spec/01-unit/002-conf_loader_spec.lua

```diff
@@ -73,16 +69,39 @@ describe("Configuration loader", function()
     assert.True(conf.plugins["hello-world"])
     assert.True(conf.plugins["another-one"])
   end)
-  it("extracts ports and listen ips from proxy_listen/admin_listen", function()
+  it("extracts flags, ports and listen ips from proxy_listen/admin_listen", function()
     local conf = assert(conf_loader())
-    assert.equal("127.0.0.1", conf.admin_ip)
-    assert.equal(8001, conf.admin_port)
-    assert.equal("127.0.0.1", conf.admin_ssl_ip)
-    assert.equal(8444, conf.admin_ssl_port)
-    assert.equal("0.0.0.0", conf.proxy_ip)
-    assert.equal(8000, conf.proxy_port)
-    assert.equal("0.0.0.0", conf.proxy_ssl_ip)
-    assert.equal(8443, conf.proxy_ssl_port)
+    assert.equal("127.0.0.1", conf.admin_listeners[1].ip)
+    assert.equal(8001, conf.admin_listeners[1].port)
+    assert.equal(false, conf.admin_listeners[1].ssl)
+    assert.equal(false, conf.admin_listeners[1].http2)
+    assert.equal("127.0.0.1:8001", conf.admin_listeners[1].listener)
+
+    assert.equal("127.0.0.1", conf.admin_listeners[2].ip)
+    assert.equal(8444, conf.admin_listeners[2].port)
+    assert.equal(true, conf.admin_listeners[2].ssl)
+    assert.equal(false, conf.admin_listeners[2].http2)
+    assert.equal("127.0.0.1:8444 ssl", conf.admin_listeners[2].listener)
+
+    assert.equal("0.0.0.0", conf.proxy_listeners[1].ip)
+    assert.equal(8000, conf.proxy_listeners[1].port)
+    assert.equal(false, conf.proxy_listeners[1].ssl)
+    assert.equal(false, conf.proxy_listeners[1].http2)
+    assert.equal("0.0.0.0:8000", conf.proxy_listeners[1].listener)
+
+    assert.equal("0.0.0.0", conf.proxy_listeners[2].ip)
+    assert.equal(8443, conf.proxy_listeners[2].port)
+    assert.equal(true, conf.proxy_listeners[2].ssl)
+    assert.equal(false, conf.proxy_listeners[2].http2)
+    assert.equal("0.0.0.0:8443 ssl", conf.proxy_listeners[2].listener)
+  end)
+  it("extracts 'off' from proxy_listen/admin_listen", function()
+    local conf = assert(conf_loader(nil, {
+          proxy_listen = "off",
+          admin_listen = "off",
+        }))
+    assert.same({}, conf.proxy_listeners)
+    assert.same({}, conf.admin_listeners)
```

Comment:
Can we also add a test for the issue described yesterday with values such as `offshore.com:80` and ensuring they do not disable the proxy? I find that we lack testing in corner cases and invalid user input in this PR